### DISCUSSION TITLE
Addressing Poor Performance: Once the PRs for Bitwarden Issue 1597 are commited, can we please get these into Vaultwarden ASAP?

### DISCUSSION THREAD

**Discussion by ccben87** [dani-garcia/vaultwarden#4404] [Ideas]
We've been having poor performance with Vaultwarden because we have lots of ciphers under the one vault. Likely around 10k ciphers. According to https://github.com/bitwarden/clients/issues/1597#issuecomment-1921377314 this will be fixed by applying the following PRs:

https://github.com/bitwarden/clients/pull/6465
https://github.com/bitwarden/clients/pull/7582
https://github.com/bitwarden/clients/pull/7585

Can we please get these implemented into Vaultwarden as soon as they are accepted by Bitwarden so that we can improve speed for people like us who have a lot of ciphers? Thanks.

**Comment by BlackDex** [comment]
We follow the updates of the Bitwarden web client, and try to at least release those into our `testing` releases after those clients are released by Bitwarden. We also first need to test these new web-vaults first with our current server code, and if that is ok, then we will release a new web-vault and after that implement it into our Vaultwarden container releases.

I would suggest to follow the commits of this project and our web-builds https://github.com/dani-garcia/bw_web_builds .

**Comment by quexten** [comment]
@ccben87 FYI the bulk encryption service upstream pr has been merged to main, making 10k cipher vaults significantly (4x and more) faster to unlock, depending on your system. For local deployments, building clients main with vaultwarden patches, and enabling the feature flag should be enough. Otherwise, on the next release of the web vault, when vaultwarden has pulled in the changes to the web vault, enabling the feature flag will enable the faster decryption on all platforms (except chrome browser extension). It will likely stay feature flagged for a few releases after that though.

**Reply by ccben87** [reply]
@quexten Thanks so much for this! I can't wait until it gets merged into a new version of BW Web Vault!

> except chrome browser extension

Don't you mean the desktop client? [eliykat](https://github.com/eliykat) said it was desktop that was not ready for this ? See here: https://github.com/bitwarden/clients/pull/6465#issuecomment-1987638081 Unless I'm missing something?

**Reply by quexten** [reply]
No, desktop client will also have the multi-threaded improvements (I implemented the required changes to make that possible on desktop in a separate PR). Only chrome browser extension (and cli) do not have them for now. I'll see if I can revisit chrome at some point.

**Reply by ccben87** [reply]
> I'll see if I can revisit chrome at some point.

If you could this would be much appreciated as we have a lot of staff that use the extension!

**Reply by ccben87** [reply]
@quexten Looks like your PR finally made it into Bitwarden Web in the new version: v2024.7.3 
Thanks again for your work. I'll be very keen to try it once this lands in VW.

**Reply by BlackDex** [reply]
To lower the expectations. It will probably take a while for it to reach Vaultwarden. There are a lot of changes on the web-vault for which we are not yet compatible and that will take time to test and implement.
